pkgpanda/__init__.py

```diff
@@ -656,18 +699,24 @@ def symlink_all(src, dest):
             # Add to the active folder
             os.symlink(package.path, os.path.join(self._make_abs("active.new"), package.name))
 
-            # Add to the environment contents
-            env_contents += "# package: {0}\n".format(package.id)
-            for k, v in package.environment.items():
-                env_contents += "{0}={1}\n".format(k, v)
-            env_contents += "\n"
+            # Add to the environment and environment.export contents
 
-            # Add to the environment.export contents
+            env_contents += "# package: {0}\n".format(package.id)
             env_export_contents += "# package: {0}\n".format(package.id)
+
             for k, v in package.environment.items():
+                env_contents += "{0}={1}\n".format(k, v)
                 env_export_contents += "export {0}={1}\n".format(k, v)
+
+            env_contents += "\n"
             env_export_contents += "\n"
 
+            # execute sysctl
+            for sysctl_var, sysctl_value in package.sysctl.items():
+                if _is_root():
+                    _set_sysctl(sysctl_value, sysctl_var)
```

Comment:
Setting sysctls should not happen here, it should happen as part of package activation. The pacakge activation is setup as an atomic swap. Everything up until the swap is non-destructive (We're just preparing some directories by making lots of symlinks), then the swap itself is atomic. The setting sysctls on the active system should come near the end of that. Making the "persistent sysctl" settings file should come fairly early on so that moving it into place can be part of the swap.